package.json

```
@@ -9,7 +9,15 @@
   "module": "./dist/fuse.esm.js",
   "unpkg": "./dist/fuse.js",
   "jsdelivr": "./dist/fuse.js",
-  "typings": "./dist/fuse.d.ts",
+  "types": "./dist/fuse.d.ts",
+  "type": "module",
+  "exports": {
+    ".": {
+      "types": "./dist/fuse.d.ts",
+      "import": "./dist/fuse.esm.js",
+      "require":  "./dist/fuse.common.js"
+    }
+  },
```

```suggestion
  "exports": {
    "types": "./dist/fuse.d.ts",
    "import": "./dist/fuse.esm.js",
    "require":  "./dist/fuse.common.js"
  },
```

The old version was invalid. You cannot just add `"type": "module"` unless Fuse itself is rewritten to ESM. Also `"type": "module"` only affects the module itself, not when the module is loaded through node_modules.

Furthermore the subgrouping of `"."` is not necessary because Fuse doesn't have subpath exports

Lastly, after including this change thus reverting `"type": "module"` you should also revert the other `.cjs` related changes.

---

Thanks for reviewing! I have updated to your suggestions.

It's been a while and I haven't checked again, locally, but I hope this would work well with Typescript + Jest

---

You'll also have to rebase it because I dropped Jest in #711 and the repo now uses Vitest

---

It should. I maintain the [Sapphire project](https://github.com/sapphiredev) and we use this structure in all projects there are it works with a variety of stacks. CJS, ESM, Jest, Vitest, etc